les/client_handler.go

```diff
@@ -232,9 +232,27 @@ func (h *clientHandler) handleMsg(p *peer) error {
 		p.fcServer.ReceivedReply(resp.ReqID, resp.BV)
 		if h.fetcher.requestedID(resp.ReqID) {
 			h.fetcher.deliverHeaders(p, resp.ReqID, resp.Headers)
-		} else {
-			if err := h.downloader.DeliverHeaders(p.id, resp.Headers); err != nil {
-				log.Debug("Failed to deliver headers", "err", err)
+		} else { // TODO(lucas): this is likely imperfect
+			h.backend.retriever.lock.RLock()
```

Comment:
I believe this is possible, seems like the odr code will try sending a request and if it fails it well keep sending to new peers, under the same request ID.
This seems fine to me, since even if a request gets answered twice it will still need to pass validation, and if both responses pass validation then we just update the header in the db twice.
Am I missing something?

Reply:
That makes sense.